yt/units/unit_object.py

```diff
@@ -1 +1,34 @@
-from unyt.unit_object import *
+# re-export all of unyt.unit_object's public interface, except for `define_unit`
+from unyt.unit_object import sympy_one, Unit, em_conversions, em_conversion_dims, NULL_UNIT
```

Comment:
Open question to reviewers:
This is the 100% backward compatible refactor, but do we actually want to re-export everything public from this external module ? I think we just really need `Unit`, and the rest can be considered implantation details ?

Reply:
I'm not sure it's technically possible to deprecate importing specific objects.
Since this PR is also not going in the direction of deprecating `yt.units` as a public namespace, I think we could try re-exposing only relevant subsets of unyt (here it would be just `Unit`) at the top level (in `yt.units`), and adapt/discuss if any complaints ensue. I will probably propose something like this early in the 4.2 dev cycle